### PR TITLE
create missing python script from notebook if it does not exist.

### PR DESCRIPTION
**Pull request by ErwanDemairy**
**PR description**

cf. the title.

**Developer Certificate Of Origin (DCO)**

By opening this merge request, you agree to the
[Developer Certificate of Origin (DCO)](https://github.com/fedbiomed/fedbiomed/blob/master/CONTRIBUTING.md#fed-biomed-developer-certificate-of-origin-dco)

This DCO essentially means that:

- you offer the changes under the same license agreement as the project, and
- you have the right to do that,
- you did not steal somebody else’s work.

**License**

Project code files should begin with these comment lines to help trace their origin:
```
# This file is originally part of Fed-BioMed
# SPDX-License-Identifier: Apache-2.0
```

Code files can be reused from another project with a compatible non-contaminating license.
They shall retain the original license and copyright mentions.
The `CREDIT.md` file and `credit/` directory shall be completed and updated accordingly.

**Guidelines for PR review**

General:

* give a glance to [DoD](https://fedbiomed.org/latest/developer/definition-of-done/)
* check [coding rules and coding style](https://fedbiomed.org/latest/developer/usage_and_tools/#coding-style)
* check docstrings (eg run `tests/docstrings/check_docstrings`)

Specific to some cases:

* update all conda envs consistently (`development` and `vpn`, Linux and MacOS)
* if modified researcher (eg new attributes in classes) check if breakpoint needs update (`breakpoint`/`load_breakpoint` in `Experiment()`, `save_state_breakpoint`/`load_state_breakpoint` in aggregators, strategies, secagg, etc.)
* if modified a component with versioning (config files, breakpoint, messaging protocol) then update the version following the rules in `common/utils/_versions.py`

### CODE REVIEW

File: scripts/run_test_mnist [hunk 1]
```diff
@@ -62,6 +62,12 @@ kpids="${kpids:+$kpids }$apids"
 echo "INFO: launching fedbiomed researcher"
 source $basedir/scripts/fedbiomed_environment researcher
 EXEC_TEST=$basedir/notebooks/101_getting-started.py
+if [ ! -f "$EXEC_TEST" ]
+then
+    EXEC_TEST_NB=${EXEC_TEST%.py}.ipynb
+    echo "converting ${EXEC_TEST_NB} to ${EXEC_TEST}"
+    jupyter nbconvert --to script "${EXEC_TEST_NB}"
```
Comment (mvesin) [inline scripts/run_test_mnist:69]: @ErwanDemairy conda environments need additional package for running `jupyter nbconvert`. Currently we get:

```
jupyter nbconvert --to script ./notebooks/101_getting-started.ipynb
...
ModuleNotFoundError: No module named 'notebook.base'
```

Comment (mvesin) [reply scripts/run_test_mnist:69]: Hi @ErwanDemairy 
This PR is still ongoing.
We also have #954 user issue that might be related to removing `notebooks/*.py`
I also realized removing `notebooks/*.py` breaks the documentation examples using script (`python ./notebooks/101_getting-started.py`) so we need to update them

Comment (ErwanDemairy) [reply scripts/run_test_mnist]: Should the notebooks in the documentation directories also be runnable by python, once nb-converted?
Currently, there are 3 notebooks including the autoreload snippet, e.g. they require ipython.
```bash
$ grep -r autorel *                  
docs/tutorials/security/non-private-local-central-dp-monai-2d-image-registration.ipynb:    "%load_ext autoreload\n",
docs/tutorials/security/non-private-local-central-dp-monai-2d-image-registration.ipynb:    "%autoreload 2"
docs/tutorials/monai/02_monai-2d-image-registration.ipynb:    "%load_ext autoreload\n",
docs/tutorials/monai/02_monai-2d-image-registration.ipynb:    "%autoreload 2"
docs/tutorials/scikit-learn/02_sklearn_sgd_regressor_tutorial.ipynb:    "%load_ext autoreload\n",
docs/tutorials/scikit-learn/02_sklearn_sgd_regressor_tutorial.ipynb:    "%autoreload 2"
```

Comment (mvesin) [reply scripts/run_test_mnist]: You can remove it also in tutorials - we don't need that snippet in tutorials !

Comment (ErwanDemairy) [reply scripts/run_test_mnist]: IMO, 
1. this conversation is solved.
2. the PR can be merged and closed.

**Review by mvesin — APPROVED** [review]

**Pull request merged**